core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java

```diff
@@ -38,14 +43,20 @@
      */
     void writeTo(StreamOutput out) throws IOException;
 
-     @FunctionalInterface
-     interface IOFunction<T, R> {
-         /**
-          * Applies this function to the given argument.
-          *
-          * @param t the function argument
-          * @return the function result
-          */
-         R apply(T t) throws IOException;
-     }
+    @Override
+    default T readFrom(StreamInput in) throws IOException {
```

Comment:
I think I *have*. My plan is to remove more of them as I cut them over to the new NamedWriteableRegistry method.